File: cs/libldt3/model/objekte/Kommunikationsdaten.cs
/*
 * Copyright 2016-2024  Christoph Brill <opensource@christophbrill.de>
 *
 * Permission is hereby granted, free of charge, to any person obtaining a copy
 * of this software and associated documentation files (the "Software"), to deal
 * in the Software without restriction, including without limitation the rights
 * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
 * copies of the Software, and to permit persons to whom the Software is
 * furnished to do so, subject to the following conditions:
 *
 * The above copyright notice and this permission notice shall be included in all
 * copies or substantial portions of the Software.
 *
 * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
 * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
 * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
 * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
 * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
 * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
 * SOFTWARE.
 */
using libldt3.attributes;
using libldt3.model;
using libldt3.model.regel.kontext;

namespace libldt3
{
    namespace model
    {
        namespace objekte
        {
            /// <summary>
            /// Hier werden alle Informationen zusammengefasst, die eine Kommunikation bspw.
            /// </summary>
            /// Mit
            /// einer Einrichtung, Firma, Arzt, einem Patienten ermöglichen.
            [Objekt(Value = "0031", Kontextregeln = new[] { typeof(K059) })]
            public class Kommunikationsdaten : Kontext
            {
                [Objekt]
                public class AlternativeElektronischePostadresse : Kontext
                {
                    public string Value;
                    [Feld(Value = "7340", Feldart = Feldart.bedingt_muss)]
                    [Regelsatz(MaxLaenge = 60)]
                    public string SpezifizierungAlternativenElektronischenPostadresse;

                }
                [Feld(Value = "7330", Feldart = Feldart.bedingt_muss)]
                [Regelsatz(MaxLaenge = 60)]
                public IList<string> Telefonnummer;
                [Feld(Value = "7331", Feldart = Feldart.bedingt_muss)]
                [Regelsatz(MaxLaenge = 60)]
                public IList<string> Mobiltelefonnummer;
                [Feld(Value = "7332", Feldart = Feldart.bedingt_muss)]
                [Regelsatz(MaxLaenge = 60)]
                public IList<AlternativeElektronischePostadresse> AlternativeElektronischePostadresse;
                [Feld(Value = "7333", Feldart = Feldart.bedingt_muss)]
                [Regelsatz(MaxLaenge = 60)]
                public IList<string> Faxnummer;
                [Feld(Value = "7335", Feldart = Feldart.bedingt_muss)]
                [Regelsatz(MaxLaenge = 60)]
                public IList<string> EMailadresse;
                [Feld(Value = "7334", Feldart = Feldart.bedingt_muss)]
                [Regelsatz(MaxLaenge = 60)]
                public IList<string> Webadresse;

            }
        }
    }
}

File: java/src/main/java/libldt3/model/objekte/Kommunikationsdaten.java
/*
 * Copyright 2016-2023  Christoph Brill &lt;opensource@christophbrill.de&gt;
 *
 * Permission is hereby granted, free of charge, to any person obtaining a copy
 * of this software and associated documentation files (the "Software"), to deal
 * in the Software without restriction, including without limitation the rights
 * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
 * copies of the Software, and to permit persons to whom the Software is
 * furnished to do so, subject to the following conditions:
 *
 * The above copyright notice and this permission notice shall be included in all
 * copies or substantial portions of the Software.
 *
 * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
 * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
 * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
 * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
 * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
 * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
 * SOFTWARE.
 */
package libldt3.model.objekte;

import java.util.List;

import libldt3.annotations.Feld;
import libldt3.annotations.Feldart;
import libldt3.annotations.Objekt;
import libldt3.annotations.Regelsatz;
import libldt3.model.Kontext;
import libldt3.model.regel.kontext.K059;

/**
 * Hier werden alle Informationen zusammengefasst, die eine Kommunikation bspw. Mit
 * einer Einrichtung, Firma, Arzt, einem Patienten ermöglichen.
 */
@Objekt(value = "0031", kontextregeln = K059.class)
public class Kommunikationsdaten implements Kontext {

    @Objekt
    public static class AlternativeElektronischePostadresse implements Kontext {
        @SuppressWarnings("unused")
        public String value;
        @Feld(value = "7340", feldart = Feldart.bedingt_muss)
        @Regelsatz(maxLaenge = 60)
        public String spezifizierungAlternativenElektronischenPostadresse;
    }

    @Feld(value = "7330", feldart = Feldart.bedingt_muss)
    @Regelsatz(maxLaenge = 60)
    public List<String> telefonnummer;
    @Feld(value = "7331", feldart = Feldart.bedingt_muss)
    @Regelsatz(maxLaenge = 60)
    public List<String> mobiltelefonnummer;
    @Feld(value = "7332", feldart = Feldart.bedingt_muss)
    @Regelsatz(maxLaenge = 60)
    public List<AlternativeElektronischePostadresse> alternativeElektronischePostadresse;
    @Feld(value = "7333", feldart = Feldart.bedingt_muss)
    @Regelsatz(maxLaenge = 60)
    public List<String> faxnummer;
    @Feld(value = "7335", feldart = Feldart.bedingt_muss)
    @Regelsatz(maxLaenge = 60)
    public List<String> eMailadresse;
    @Feld(value = "7334", feldart = Feldart.bedingt_muss)
    @Regelsatz(maxLaenge = 60)
    public List<String> webadresse;

}
